app/src/main/java/camp/nextstep/DispatcherServlet.java

```diff
@@ -1,51 +1,75 @@
 package camp.nextstep;
 
+import com.interface21.webmvc.servlet.ModelAndView;
+import com.interface21.webmvc.servlet.mvc.tobe.AnnotationHandlerMapping;
+import com.interface21.webmvc.servlet.mvc.tobe.ControllerHandlerAdapter;
+import com.interface21.webmvc.servlet.mvc.tobe.HandlerAdapter;
+import com.interface21.webmvc.servlet.mvc.tobe.HandlerMapping;
 import jakarta.servlet.ServletException;
 import jakarta.servlet.http.HttpServlet;
 import jakarta.servlet.http.HttpServletRequest;
 import jakarta.servlet.http.HttpServletResponse;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import com.interface21.webmvc.servlet.view.JspView;
+
+import java.util.ArrayList;
+import java.util.List;
 
 public class DispatcherServlet extends HttpServlet {
 
     private static final long serialVersionUID = 1L;
     private static final Logger log = LoggerFactory.getLogger(DispatcherServlet.class);
 
-    private ManualHandlerMapping manualHandlerMapping;
-
+    private List<HandlerMapping> handlerMappings = new ArrayList<>();
+    private List<HandlerAdapter> handlerAdapters = new ArrayList<>();
     public DispatcherServlet() {
     }
 
+    public void addHandlerAdapter(HandlerAdapter handlerAdapter) {
+        handlerAdapters.add(handlerAdapter);
+    }
+
+    public void addHandlerMapping(HandlerMapping handlerMapping) {
+        handlerMappings.add(handlerMapping);
+    }
+
     @Override
     public void init() {
-        this.manualHandlerMapping = new ManualHandlerMapping();
-        this.manualHandlerMapping.initialize();
+        // 메뉴얼 핸들러 추가
+        ManualHandlerMapping manualHandlerMapping = new ManualHandlerMapping();
+        manualHandlerMapping.initialize();
+        addHandlerMapping(manualHandlerMapping);
+
+        // 어노테이션 핸들러 추가
+        AnnotationHandlerMapping annotationHandlerMapping = new AnnotationHandlerMapping("camp.nextstep.controller");
+        annotationHandlerMapping.initialize();
+        addHandlerMapping(annotationHandlerMapping);
+
+        // 핸들러 어댑터 추가
+        addHandlerAdapter(new ControllerHandlerAdapter());
     }
 
     @Override
     protected void service(final HttpServletRequest request, final HttpServletResponse response) throws ServletException {
         final String requestURI = request.getRequestURI();
         log.debug("Method : {}, Request URI : {}", request.getMethod(), requestURI);
 
+        Object handler = handlerMappings.stream()
+                .map(handlerMapping -> handlerMapping.getHandler(request))
+                .findFirst()
+                .orElseThrow(() -> new IllegalArgumentException("유효하지 않은 URI : " + request.getRequestURI()));
+
+        HandlerAdapter handlerAdapter = handlerAdapters.stream()
+                .filter(adapter -> adapter.supports(handler))
+                .findFirst()
+                .orElseThrow(() -> new IllegalArgumentException("유효하지 않은 핸들러 : " + handler.getClass().getName()));
```

Comment:
일급 컬렉션을 사용한다면 다음과 같이 작성될 수 있을 것 같아요. 
```suggestion
        Object handler = handlerMappings.getHandler(request);
        HandlerAdapter handlerAdapter = handlerAdapters.getHandlerDapter(handler);
        
        ModelAndView mv = handlerAdapter.handle(request, response, handler);
        mv.render(request, response);
```